### ISSUE TITLE
Speakers not working on Acer Chromebook 317

### ISSUE THREAD

**Issue by defitzwa** [WeirdTreeThing/chromebook-linux-audio#11]
I am using Fedora 38 (linux kernel 6.3.8-200.fc38.x86_64) with chromebook-linux-audio on an Acer Chromebook 317.

It is my understanding that this is a Jasperlake Chromebook with a Magpie board, so when searching for a solution, I came across the following thread from a familiar user:
https://github.com/thesofproject/sof/issues/7162

If I read it correctly, it sounds like there was a resolution in the end. But maybe not yet rolled out? Are there any instructions for how I can get this working on my machine?

**Comment by WeirdTreeThing** [comment]
My script will install the updated topologies for you, so it should just work.

**Comment by defitzwa** [comment]
I ran the script before posting (apologies for not stating that clearly). The audio through the speakers is not working, but the audio through the headset is working after running the script. Before running the script, the headset audio was not working either.

It is my understanding that the script installed some or all of the files shown under /lib/firmware/intel/sof and /lib/firmware/intel/sof-tplg/

In case it is helpful, I have attached the outputs of 
ls -al /lib/firmware/intel/sof
[lib_firmware_intel_sof.txt](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11930301/lib_firmware_intel_sof.txt)

and 
ls -al /lib/firmware/intel/sof-tplg/
[lib_firmware_intel_sof-tplg.txt](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11930302/lib_firmware_intel_sof-tplg.txt)


Is there anything else I can check that would help in troubleshooting the issue?

**Comment by WeirdTreeThing** [comment]
Send the output of `aplay -l`.

**Comment by defitzwa** [comment]
Please see attached: 
[aplay -l.txt](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11931593/aplay.-l.txt)

**Comment by WeirdTreeThing** [comment]
Run this and upload the file it creates 
[audio-debug.sh.txt](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11931694/audio-debug.sh.txt)

**Comment by defitzwa** [comment]
Here is the log file:

[audio-debug.log](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11931739/audio-debug.log)

**Comment by WeirdTreeThing** [comment]
So topology and ucm both look fine. Send your dmesg log with `sudo dmesg > dmesg.log`.

**Comment by defitzwa** [comment]
There do appear to be some errors in this one: 

[dmesg.log](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/11931774/dmesg.log)

**Comment by WeirdTreeThing** [comment]
Those errors are normal. Try running `aplay -d plughw:0,0 /usr/share/sounds/alsa/Front_Center.wav`

**Comment by defitzwa** [comment]
Konsole didn't like the argument -d:

aplay -d plughw:0,0 /usr/share/sounds/alsa/Front_Center.wav
aplay: main:659: invalid duration argument 'plughw:0,0'

It was happy with an uppercase D though:
aplay -D plughw:0,0 /usr/share/sounds/alsa/Front_Center.wav
Playing WAVE '/usr/share/sounds/alsa/Front_Center.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Mono

I'm guessing this is meant to play sound. Nothing came out of the onboard speakers though.

**Comment by WeirdTreeThing** [comment]
Oh sorry I wrote the command wrong, but yeah the sound should be played out of the speakers. It looks like a kernel/firmware issue though. Anything new in dmesg after running that command?

**Comment by defitzwa** [comment]
Nope, only addition is from where a tried and failed to run a diff command

**Comment by defitzwa** [comment]
Apologies. I was not reading the output correctly. Yes, there are a couple new entries in dmesg that most likely do not have anything to do with me using the wrong argument in a diff.

[ 6990.115453] perf: interrupt took too long (2510 > 2500), lowering kernel.perf_event_max_sample_rate to 79000
[ 7439.327688] PDLOG 2023/07/03 00:11:35.722 P1 Disconnected

**Comment by WeirdTreeThing** [comment]
Those aren't related to audio

**Comment by defitzwa** [comment]
Yes, I came to that conclusion after googling them. Looks like it is just logging CPU scaling and then logging when I disconnect from wall power. Any other things I can try or look at? You mentioned the kernel/firmware. I would prefer to stick with Fedora but I suppose I could try a different distribution in the name of science.

**Comment by WeirdTreeThing** [comment]
Make sure your system is fully up-to-date. Run the audio script again and see if it still doesn't work.

**Comment by defitzwa** [comment]
Sorry for the slow response. I was traveling last week and didn't want to run the update through my phone hotspot. 

System has been updated. Re-pulled and re-ran the script. Does not appear to have changed behavior. Offboard audio works but onboard does not.

**Comment by WeirdTreeThing** [comment]
Ok so most likely a UCM issue. JSL hasn't been the best supported for this for a while. Run `cat /sys/class/dmi/id/product_sku` and send the output.

**Comment by defitzwa** [comment]
The output was simply:

sku720896

**Comment by WeirdTreeThing** [comment]
I just updated the UCM, run the script and try again.

**Comment by defitzwa** [comment]
I re-pulled https://github.com/WeirdTreeThing/chromebook-linux-audio and reran the setup-audio script, but did not see a change in audio behavior. 

I reran `cat /sys/class/dmi/id/product_sku`. the ouput is still `sku720896`

I see there is another sizable Fedora update that became available today. I will try installing that and rerunning the script again.

**Comment by WeirdTreeThing** [comment]
The SKU wont change. Run [this script](https://cdn.discordapp.com/attachments/1121994891950559283/1126689828063412234/audio-debug.sh) and send the output of it as an attachment.

**Comment by defitzwa** [comment]
Output is attached 
[audio-debug.log](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/12424987/audio-debug.log)

**Comment by WeirdTreeThing** [comment]
Can you kill your audio server (`systemctl --user stop pipewire.socket && systemctl --user stop pipewire-pulse.socket`) and try again?

**Comment by defitzwa** [comment]
Here is the output of the audio-debug script after killing the audio server: 
[audio-debug.log](https://github.com/WeirdTreeThing/chromebook-linux-audio/files/12425050/audio-debug.log)

**Comment by WeirdTreeThing** [comment]
That all looks normal, not too sure why your speakers aren't working then. I'm gonna have someone else with the same device test their speakers, I know it worked for them before so a recent change might have broken it.

**Comment by defitzwa** [comment]
Sounds like a good plan, thank you for the help.

**Comment by WeirdTreeThing** [comment]
Forgot about this issue. I made some improvements to the magpie ucm a few weeka ago, try again.

**Comment by defitzwa** [comment]
I re-cloned the repository and re-ran the script, then rebooted. The onboard audio is still stubbornly silent. Did the other person with the same hardware respond?

**Comment by WeirdTreeThing** [comment]
Unfortunately, that person had some issues with their hardware.

**Comment by defitzwa** [comment]
Bummer. The mystery persists.
